sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -85,6 +86,23 @@ public Response Response
         /// </summary>
         public TimeSpan? NetworkTimeout { get; set; }
 
+        internal int RetryNumber { get; set; }
+
+        internal Exception? LastException { get; set; }
+
+        internal DateTimeOffset OperationStartTime { get; set; }
+
+        /// <summary>
+        /// The processing context for the message.
+        /// </summary>
+        public ProcessingContext ProcessingContext
+        {
+            get => _processingContext ??= new ProcessingContext(this);
```

Comment:
The point here is to lazily allocate the ProcessingContext.

Reply:
doesn't it need to be thread safe? BTW, if we make it a struct, then we don't have to lazy allocate it and no need for thread safety.

Reply:
No longer lazily allocating.